common/flatpak-dir.c

```diff
@@ -10766,34 +10766,41 @@ flatpak_dir_remote_fetch_summary (FlatpakDir   *self,
     }
   else
     {
-      g_debug ("Fetching summary file for remote ‘%s’", name_or_uri);
-      if (only_cached)
-        {
-          g_autofree char *sig_name = g_strconcat (name_or_uri, ".sig", NULL);
-          g_autoptr(GFile) summary_cache_file = flatpak_build_file (self->cache_dir, "summaries", name_or_uri, NULL);
-          g_autoptr(GFile) summary_sig_cache_file = flatpak_build_file (self->cache_dir, "summaries", sig_name, NULL);
-          g_autoptr(GMappedFile) mfile = NULL;
-          g_autoptr(GMappedFile) sig_mfile = NULL;
+      g_autofree char *sig_name = g_strconcat (name_or_uri, ".sig", NULL);
+      g_autoptr(GFile) summary_cache_file = flatpak_build_file (self->cache_dir, "summaries", name_or_uri, NULL);
+      g_autoptr(GFile) summary_sig_cache_file = flatpak_build_file (self->cache_dir, "summaries", sig_name, NULL);
+      g_autoptr(GMappedFile) mfile = NULL;
+      g_autoptr(GMappedFile) sig_mfile = NULL;
 
-          mfile = g_mapped_file_new (flatpak_file_get_path_cached (summary_cache_file), FALSE, NULL);
-          if (mfile == NULL)
-            {
-              g_set_error (error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_CACHED,
-                           _("No summary cached for remote '%s'"), name_or_uri);
-              return FALSE;
-            }
+      g_debug ("Attempting to load cached summary file for remote ‘%s’", name_or_uri);
 
+      mfile = g_mapped_file_new (flatpak_file_get_path_cached (summary_cache_file), FALSE, NULL);
+      if (mfile == NULL && only_cached)
+        {
+          g_set_error (error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_CACHED,
+                       _("No summary cached for remote '%s'"), name_or_uri);
+          return FALSE;
+        }
+      else if (mfile != NULL)
+        {
           sig_mfile = g_mapped_file_new (flatpak_file_get_path_cached (summary_sig_cache_file), FALSE, NULL);
 
           summary = g_mapped_file_get_bytes (mfile);
           if (sig_mfile)
             summary_sig = g_mapped_file_get_bytes (sig_mfile);
         }
-      else if (!ostree_repo_remote_fetch_summary (self->repo, name_or_uri,
-                                                  &summary, &summary_sig,
-                                                  cancellable,
-                                                  error))
-        return FALSE;
+      else if (!only_cached)
```

Comment:
This changes the behaviour. The previous behaviour was: Unless only_cached is specified we always try to get an up-to-date summary file, but downloading the summary.sig file and comparing that to the on-disk version. This is presumably fast because the signature is small.

With this change we will never update the summary, which is not right.

The addition of only_cached is based on the request in https://github.com/flatpak/flatpak/issues/2817